### PR TITLE
Replace wallet connection logic with Rainbow Kit

### PR DESCRIPTION
**Pull request by raulrpearson**
I think I replaced all the wallet and chain related components with the Rainbow Kit counterparts. I didn't customize [theme](https://www.rainbowkit.com/docs/custom-theme) and [wallet list](https://www.rainbowkit.com/docs/custom-wallet-list) much because those APIs are flagged as unstable and we still have a while to go to mainnet.

The stake/unstake controls on the CreatorCard component don't update correctly after staking (even though the staking flow seems to work, but I think this is not related to the Rainbow Kit change. I was able to reproduce on develop. Can you have a look at it and confirm? If so, I'd say we can open a new issue to have a look at it.

Let me know what you think or if I missed something. Feel free to merge if everything looks good (although keep in mind it might need rebasing if some other PR is merged first). Thanks!

### CODE REVIEW

File: packages/frontend/lib/wagmiClient.ts
Comment (wottpal) [inline packages/frontend/lib/wagmiClient.ts:28]: TL;DR Opened an issue here that describes everything: https://github.com/rainbow-me/rainbowkit/issues/631. I would wait for their reply before merging (they are generally super fast)

**Review by wottpal — CHANGES_REQUESTED** [review]
As mentioned in the comment, I would wait for approval until some comments on https://github.com/rainbow-me/rainbowkit/issues/631.

Two more things I would suggest:
* Update wagmi to `^0.6.0` right away (no breaking changes for us)
* As initially the connect-button in the navbar was replaced with a button that links to someones profile, Rainbowkit obviously doesn't do that. I would recommend to do that in the hero in `index.tsx` now instead. Otherwise it's quite complicated for the user to find their own profile-pages. An option Could be to check whether a logged-in address is in the `creators` array and then instead show a "Go to my Profile" link [here](https://github.com/yieldgate/yieldgate/blob/0239bb791cf1df8a5036000c024aaab15b98c1d8/packages/frontend/pages/index.tsx#L37), if true.

File: packages/frontend/lib/wagmiClient.ts
Comment (wottpal) [reply packages/frontend/lib/wagmiClient.ts:28]: Update: They do not seem to answer. As I don't want this to be blocking us, I think we can accept it the way you suggested for this PR and change when they reply :)

What do you think regarding the other two points I suggested?

Comment (raulrpearson) [reply packages/frontend/lib/wagmiClient.ts:28]: I agree with both and with moving forward. I'll work on it this afternoon.

**Comment by raulrpearson** [comment]
> * As initially the connect-button in the navbar was replaced with a button that links to someones profile, Rainbowkit obviously doesn't do that. I would recommend to do that in the hero in `index.tsx` now instead. Otherwise it's quite complicated for the user to find their own profile-pages. An option Could be to check whether a logged-in address is in the `creators` array and then instead show a "Go to my Profile" link [here](https://github.com/yieldgate/yieldgate/blob/0239bb791cf1df8a5036000c024aaab15b98c1d8/packages/frontend/pages/index.tsx#L37), if true.

@wottpal, since we'll be removing profiles in the coming pivot, should we just skip this?

If you'd prefer to keep some link somewhere, I think we should find a way to included it persistently in the nav bar rather than the homepage hero. In it's simplest form it could just be a "My Profile" link on the top right:

nav-profile-link

What do you think? Skipping, nav bar, hero or something else?

**Comment by wottpal** [comment]
I would say we should go for the very simple "My Profile" link you added in your screenshot if the user is logged-in :)
If we just skip it the profiles basically become unreachable when we merge in `main` (even though i don't think anyone is actively using our product this would still somehow feel like a regression from the hackathon). What do you think?

**Comment by raulrpearson** [comment]
Let's do it, it's a simple enough change 👍

**Review by wottpal — APPROVED** [review]
lgtm

Also, the rainbow-kit issue regarding mainnet is already resolved in a pr. I assume they merge it in the next days, so I can make it a part of #37.

**Comment by raulrpearson** [comment]
Closes #24

**Pull request merged**
